great_expectations/core/validation_config.py

```diff
@@ -194,6 +194,22 @@ def _decode_data(cls, data_dict: dict) -> BatchConfig:
 
         return batch_definition
 
+    @property
+    def active_batch_id(self) -> str | None:
+        validator = self._build_validator()
+        return validator.active_batch_id
```

Comment:
This feels a bit weird but I need a way to grab the `active_batch_id` during the `run` call. Perhaps this property doesn't make sense outside of that context (since "active" denotes something in progress).

An alternative would be grabbing this ID during the run call and updating a private attr?

Reply:
This is global state that needs to exist but only when we are inside a `run` call?
That seems like we need to redesign something but likely don't have to the time to do it now.
This seems strange because we need create a new validator each time and we should only need the validator in `run` and we should be using the same validator there as here.

For now, I would use your alternative suggestion and set this to `None` when not in a run call. That is, wrap run in a try/finally block and set the private attr after creating the validator there and reset it to None in the `finally` block. I would also add a comment explaining what this global state is.